### PR TITLE
Fixes a bug that can crash the server if two supermatter crystals smash into eachother

### LINKED ISSUE #7705: SUPERMATTER CRYSTAL ULTIMATE SERVER CRASH
So apparently ~10 supermatter crystals simultaneously delaminating is capable of crashing BOTH servers at once
I'm not sure whether this is an actual bug caused by interaction of multiple crystals popping or just a case of too many server resources in use
If it's the former, maybe it should be fixed
If it's the latter, oh well just don't spawn ten fucking full crystals

### PR DESCRIPTION
**Pull request by ghost**
I'm a big dumb baby who doesn't edit the OP to reflect knowledge of bugs my PRs fix and autoclose them
Fixes #7705

### CODE REVIEW

**Comment by Exxion** [comment]
Fixes #7705

File: code/modules/supermatter/supermatter.dm [hunk 1]
```diff
@@ -99,18 +99,18 @@
 	. = ..()
 
 /obj/machinery/power/supermatter/proc/explode()
-	var/turf/turff = get_turf(src)
+	if(universe.name != "Supermatter Cascade")
```
Comment (PJB3005) [inline code/modules/supermatter/supermatter.dm]: Use an istype instead.

Comment (ghost) [inline code/modules/supermatter/supermatter.dm]: fixed

File: code/modules/supermatter/supermatter.dm [hunk 2]
```diff
@@ -99,18 +99,18 @@
 	. = ..()
 
 /obj/machinery/power/supermatter/proc/explode()
-	var/turf/turff = get_turf(src)
+	if(istype(universe,/datum/universal_state/supermatter_cascade))
```
Comment (ComicIronic) [inline code/modules/supermatter/supermatter.dm]: So, if we're already in a cascade, you send us into a cascade, and the supermatter only actually explodes if that happens?

Comment (ghost) [inline code/modules/supermatter/supermatter.dm]: whoops, missed a !

**Comment by ghost** [comment]
ready to merge

**Comment by Iamgoofball** [comment]
qdel(universe)

**Pull request merged**